test/test_closed_streams.py

```diff
@@ -89,6 +89,31 @@ def test_closed_stream_resets_further_frames(self, frame_factory):
         assert not events
         assert c.data_to_send() == rst_frame.serialize() * 3
 
+    def test_end_closed_stream_resets_further_frames(self, frame_factory):
+        """
+        A stream that is closed can receive further frames: it simply sends
+        RST_STREAM for it.
+        """
+        c = h2.connection.H2Connection(client_side=False)
+        c.receive_data(frame_factory.preamble())
+        c.initiate_connection()
+
+        f = frame_factory.build_headers_frame(
+            self.example_request_headers, flags=['END_STREAM']
+        )
+        c.receive_data(f.serialize())
+        c.send_headers(
+            stream_id=1,
+            headers=self.example_response_headers,
+            end_stream=True
+        )
+        c.clear_outbound_data_buffer()
+
+        data_frame = frame_factory.build_data_frame(b'hi there')
+
+        with pytest.raises(h2.exceptions.ProtocolError):
+            c.receive_data(data_frame.serialize())
+
```

Comment:
Of course, that's because you don't expect a `RST_STREAM` frame, you expect a `GOAWAY` frame. You'll need to change the docstring too. =)